### DISCUSSION TITLE
Integration scenario: online self service?

### DISCUSSION THREAD

**Discussion by mbingel** [Digitaler-Impfnachweis/certification-apis#71] [Q&A]
Hallo zusammen,

die Dokumentation geht aktuell von zwei Szenarien aus, bei denen in beiden Fällen dem Impfling das Zertifikat quasi live durch die Organisation ausgehändigt wird:
1. Medical Practitioner office 
2. Vaccination Center

Unsere Impf-Management Lösung ist mehr Self-Service für den Impfling und wir hatten folgenden Ablauf geplant:
1. Der Impfling bucht seine Termine online und geht zur Impfung.
2. Das Impf-Zentrum vermerkt die erfolgreiche Impfung im System (bisher genutzt für das Reporting an das RKI).
3. Der Impfling kann das digitale Zertifikat einfach im Online-Portal abrufen, es wird on-demand erzeugt.

Ist dies mit Szenario 2 abbildbar, insbesondere da Szenario 2 von Hardware-Tokens ausgeht? Müssten die dann ins Rechenzentrum unserer SaaS-Lösung? Für jeden Impfzentrum-Mandanten ein eigener Token?
("Software solution is required to have a strong user authentication based on BSI req. like hardware-tokens and user credentials")

**Comment by pklewing** [comment]
+1 - desweiteren: wie ist der Prozess des Whitelisting der Impfzentren (Szenario 2: Vaccination Center), und werden die Integrationen von externen SaaS-Lösungen mit dem DGC Service validiert und zertifiziert?

+1 - also: what is the process for whitelisting vaccination centers (scenario 2), and is validation/certification required for 3rd party SaaS solutions that integrate with the DGC service?

**Comment by ManuelB** [comment]
Hi guys,
we implemented the medical practitioner variant. Video can be seen here:
https://youtu.be/C0KfzxP-HzU

Source is here:
https://github.com/ere-health/ere-ps-app/tree/video/covid-19-ru-demo

In this scenario you need a so called connector including a SMC-B in your local network. The SMC-B authenticates the practice.

An overview about these components can be found here:
https://github.com/ere-health/architecture/blob/main/Telematik.md
